src/skdh/context/motion_detect.py

```diff
@@ -38,42 +58,301 @@ def __init__(self, filter_cutoff=6, cov_threshold=0.022558169349678834):
         self.cov_threshold = cov_threshold
 
     @handle_process_returns(results_to_kwargs=False)
-    def predict(self, accel, fs, **kwargs):
+    def predict(self, time, accel, fs=None, **kwargs):
         """
-        predict(accel, fs)
+        predict(time, accel, fs)
 
         Function to detect periods of motion from an accelerometer signal.
 
         Parameters
         ----------
+        time : numpy.ndarray
+            (N, ) array of unix timestamps, in seconds.
         accel : numpy.ndarray
-            (N, 3) array of acceleration, in units of 'g', collected at 20hz.
+            (N, 3) array of acceleration, in units of 'g'.
         fs : float
-            Sampling frequency.
+            Sampling rate. Default None. If not provided, will be inferred.
 
         Returns
         -------
         results : dict
-            Results dictionary including detected motion (1s rolling) and raw values of rolling CoV.
+            Results dictionary including detected motion (1s rolling) and raw values of 1s rolling CoV.
 
         """
+        # check input requirements are met
+        time, accel, fs = self._check_input(time, accel)
+
         # Vector Magnitude
         vmag = np.linalg.norm(accel, axis=1)
 
         # Low-pass filter the accelerometer vector magnitude signal to remove high frequency components
+        # Note: iirfilter with filtfilt used to match legacy implementation
         wn = self.filter_cutoff * 2 / fs
         [b, a] = iirfilter(self.filter_cutoff, wn, btype="lowpass", ftype="butter")
         vmag_filt = filtfilt(b, a, vmag)
 
         # Calculate the 1s rolling coefficient of variation
-        rolling_mean = moving_mean(a=vmag_filt, w_len=int(fs), skip=1)
-        rolling_std = moving_sd(a=vmag_filt, w_len=int(fs), skip=1)
-        rolling_cov = rolling_std / rolling_mean
+        # Note: added a non-zero term in denominator to prevent divide by zero errors
+        rolling_std, rolling_mean = moving_sd(a=vmag_filt, w_len=int(fs), skip=1)
+        rolling_cov = rolling_std / (rolling_mean + 1e-12)
 
-        # Detect CoV values about given movement threshold
+        # Detect CoV values above movement threshold
         movement = rolling_cov > self.cov_threshold
 
-        # compile results
+        # Group results
         results = {"movement_detected": movement, "rolling_1s_cov": rolling_cov}
```

Comment:
does this not need to be used by future steps? If it does need to be re-used, suggest creating a array of start-stop indices. Either way if its going to be re-used, then `results_to_kwargs` should be set to `True`